src/Blazor.FlexGrid/DataSet/DataSetExtensions.cs

```diff
@@ -57,5 +60,107 @@ public static string PageInfoText(this ITableDataSet tableDataSet)
+            if (groupingOptions.IsGroupingActive)
+            {
+                try
+                {
+
+                    var param = Expression.Parameter(typeof(TItem));
+                    var propertyOrField = Expression.PropertyOrField(param, groupingOptions.GroupedProperty.Name);
+
+                    var callToString = Expression.Call(propertyOrField,
+                        propertyOrField.Type.GetMethods().FirstOrDefault(m => m.Name == "ToString"));
+
+                    var callExprReturnLabel = Expression.Label(typeof(string));
+
+                    Expression<Func<TItem, string>> keyExpression;
+
+                    if (CanAcceptNullValueExpression(propertyOrField.Type))
+                    {
+                        var test = Expression.Equal(propertyOrField, Expression.Constant(null, propertyOrField.Type));
+                        var returnIfTrue = Expression.Return(callExprReturnLabel, Expression.Constant(null, typeof(string)));
+                        var returnIfFalse = Expression.Return(callExprReturnLabel, callToString);
+                        var callAndNullCheckExpr = Expression.IfThenElse(
+                            test, returnIfTrue, returnIfFalse);
+
+
+                        var lambdaBodyBlock = Expression.Block(callAndNullCheckExpr,
+                            Expression.Label(callExprReturnLabel, Expression.Constant(null, typeof(string))));
+                        keyExpression = Expression.Lambda<Func<TItem, string>>(lambdaBodyBlock, param);
+                    }
+                    else
+                    {
+                        keyExpression = Expression.Lambda<Func<TItem, string>>(callToString, param);
+                    }
+
+                    return source.GroupBy(keyExpression)
+                                .Select(grp => new GroupItem<TItem>(grp.Key, grp));
+
+                    
+                }
+                catch(Exception ex)
+                {
+                    throw ex;
+                }
+            }
+            else
+            {
+                return null;
+            }
+
+        }
+
+        private static bool CanAcceptNullValueExpression(Type type)
+        {
+            try
+            {
+                Expression.Constant(null, type);
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        public static void ToggleGroupRow<TItem>(this ITableDataSet tableDataSet, object groupItemKey)
+        {
+            var keyEqualityComparer = new GroupingKeyEqualityComparer();
+            var groupItemToToggle = tableDataSet.GroupedItems.FirstOrDefault(item => keyEqualityComparer.Equals(item.Key, groupItemKey) );
+            groupItemToToggle.IsCollapsed = !groupItemToToggle.IsCollapsed;
+
+            tableDataSet.GroupedItems = tableDataSet.GroupedItems.Select(item => !keyEqualityComparer.Equals(((GroupItem<TItem>)item).Key, groupItemKey)
```

Comment:
If you change the type of `GroupedItems ` from `IEnumerable` to `IList` you can omit this code.
Also in `TableDataSet` you have to use
 `GroupedItems = ApplyFiltersWithGroupingToQueryable(source).OfType<GroupItem>().ToList();`